src/test/spec/unified_runner/operation.rs

```diff
@@ -2120,12 +2120,48 @@ impl TestOperation for Close {
     ) -> BoxFuture<'a, Result<Option<Entity>>> {
         async move {
             let mut entities = test_runner.entities.write().await;
-            let cursor = entities.get_mut(id).unwrap().as_mut_cursor();
-            let rx = cursor.make_kill_watcher().await;
-            *cursor = TestCursor::Closed;
-            drop(entities);
-            let _ = rx.await;
-            Ok(None)
+            let target_entity = entities.get(id).unwrap();
+            match target_entity {
+                Entity::Client(_) => {
+                    let mut client = entities.get_mut(id).unwrap().as_mut_client();
+                    let closed_client_topology_id = client.topology_id;
+                    client.client = None;
+
+                    let mut entities_to_remove = vec![];
+                    for (key, value) in entities.iter() {
+                        match value {
+                            // skip clients so that we don't remove the client entity itself from
+                            // the map: we want to preserve it so we can
+                            // access the other data stored on the entity.
```

Comment:
What data do we need to access on the client entity after its been closed? And is this also why we set the inner client to `None` instead of just removing it?

Reply:
the `ClientEntity` stores:
*  the `EventHandler` and corresponding `EventObserver` for the client. no tests I've added here actually need the ability to inspect e.g. command monitoring events produced for a client after it was closed, since these tests mostly just use tracing events, but it seems plausible that would be needed in the future. (one use case: it would be good to have a spec test that checks drivers actually send `endSessions` when a client is closed, and we'd probably assert that via a `close` operation + command monitoring event.) 
* the `topology_id` (used to filter on tracing events and get those for this particular entity): currently used to make the tracing assertions. 

and yes, that's why we only get rid of the inner client and not the whole entity.